### ISSUE TITLE
ID token returned from auth emulator fails both verifyIdToken() and createSessionCookie()

### ISSUE THREAD

**Issue by Lootwig** [firebase/firebase-tools#5821]
### [REQUIRED] Environment info

**firebase-tools:** 11.30.0

**Platform:** macOS Ventura 13.2.1

**firebase**: 9.21.0
**firebase-admin**: 11.5.0

### [REQUIRED] Test case

Have the auth emulator running, and FIREBASE_AUTH_EMULATOR_HOST=localhost:9099 set.

### [REQUIRED] Steps to reproduce

```
// on client
auth = getAuth()
result = await signInWithPopup(auth, new GoogleAuthProvider().addScope('email'))
idToken = await result.user.getIdToken()

// on server
adminAuth.verifyIdToken(idToken)
// or:
adminAuth.createSessionCookie(idToken)
```

### [REQUIRED] Expected behavior

Doesn't throw

### [REQUIRED] Actual behavior

Throws either

`Error: Firebase ID token has no "kid" claim. See https://firebase.google.com/docs/auth/admin/verify-id-tokens for details on how to retrieve an ID token.`

or

`Error: The provided ID token is not a valid Firebase ID token.`

**Comment by Lootwig** [comment]
https://github.com/Lootwig/firebase-emulator

* Clone
* `npm install`
* create `.env` with
  * `PUBLIC_FIREBASE_CONFIG={...} # your project config JSON`
  * `FIREBASE_SERVICE_ACCOUNT={...} # your service account JSON`
* `firebase init`
* `firebase emulators:start`
* `npm dev`
* Open localhost:5173, click sign in, create arbitrary account, observe server process error output containing

```
errorInfo: {
    code: 'auth/invalid-id-token',
    message: 'The provided ID token is not a valid Firebase ID token.'
  },

```

**Comment by spencerchubb** [comment]
I am having the same bug. I am call getIdToken() from the client side, and then calling VerifyIdToken() on a Go server with the admin SDK. It seems calling getIdToken() from the Firebase Emulator produces an invalid token.

**Comment by Lootwig** [comment]
@spencerchubb setting the emulator host in your `.env` file does not work, try running the start command like this:

`export FIREBASE_AUTH_EMULATOR_HOST=localhost:9099; firebase emulators:start`

Does that solve it for you too?

**Comment by spencerchubb** [comment]
Thanks for the idea, but I have located the issue.

I had to update the Go admin sdk to firebase.google.com/go/v4 because it appears the previous version did not have the feature to check whether the FIREBASE_AUTH_EMULATOR_HOST env variable is set.

**Comment by xml** [comment]
FWIW... I was getting the same error about a missing "kid" claim because I originally had my `FIREBASE_AUTH_EMULATOR_HOST` set to port 4000 (`127.0.0.1:4000`), the admin dashboard port. Whereas the console where I launched the emulator very clearly and helpfully states that the HOST:Port is actually `127.0.0.1:9099`. As soon as I changed that, I suddenly have a happy and successful `verifyIdToken()` function.
